modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java

```diff
@@ -1,13 +1,16 @@
 package org.testcontainers.containers;
 
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
 import org.testcontainers.containers.wait.strategy.Wait;
 import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
 import org.testcontainers.utility.DockerImageName;
 
+import java.time.Duration;
+
 /**
  * This container wraps Apache Pulsar running in standalone mode
  */
-public class PulsarContainer extends GenericContainer<PulsarContainer> {
+public class PulsarContainer<SELF extends PulsarContainer<SELF>> extends GenericContainer<SELF> { {
```

Comment:
could you please explain this change? The API is fluent already, even without `public SELF withFunctionsWorker`